torch/csrc/autograd/custom_function.h

```diff
@@ -18,19 +19,33 @@ TORCH_API variable_list _wrap_outputs(
 // forward() can take as many arguments as you want and should return a
 // variable list. Use of any direct Variable arguments will be registered in
 // the graph but no vectors/sets or any other data structures will be traversed.
+// It should take an AutogradContext* as the first argument. Variables can be
+// saved in the ctx using save_for_backward() and other data can be saved in the
+// map ctx.save in the form of <std::string, at::IValue> pairs.
 //
-// backward() will be given a variable list containing as many Variables as
-// there were outputs from forward. It should return as many Variables as there
-// were inputs with each of them containing the gradient w.r.t. its
-// corresponding input
+// backward() should take an AutogradContext* and a variable list containing as
+// many Variables as there were outputs from forward as arguments. It should
+//  return as many Variables as there were inputs with each of them containing
+// the gradient w.r.t. its corresponding input. Variables saved in forward can
+// be accessed with ctx->get_saved_variables() and other saved data can be
+// accessed from ctx->save.
 //
 // For example:
 // class MyFunction : public CFunction<MyFunction> {
 //   public:
-//   static variable_list forward(AutogradContext *ctx, int n, Variable var);
+//   static variable_list forward(AutogradContext *ctx, int n, Variable var) {
```

Comment:
It would be very nice if the example also showed where and when to use `mark_dirty`  (and `mark_non_differentiable` but this one is more obvious).